test/libsolidity/syntaxTests/inlineAssembly/invalid/enum_value_assign.sol

```diff
@@ -0,0 +1,10 @@
+contract C {
+	enum A { X, Y, Z }
+	function f() public pure {
+		assembly {
+			A.X := 2
```

Comment:
Can you also add a test for `A.` (invalid syntax) and `A.U` (missing enum) ?

Reply:
Well, *should* that be the behaviour? Currently it only considers the identifier as reference to an enum value, if it matches exactly, otherwise it's a regular identifier. So ``A``, ``A.`` and ``A.U`` will trigger "Identifier not found" (I'll add test cases in any case), but this is valid:

```
contract C {
  enum E { X, Y, Z }
  function f() public pure returns (uint a) {
    assembly {
      let E.A := 34
       a := E.A
    }
  }
}
```

and also the following is valid (that's the same for constants - that's why I mentioned we might want to look into shadowing in general):


```
contract C {
  enum E { X, Y, Z }
  function f() public pure {
    uint a = 0; uint b = 0;
    assembly {
      let E.X := 34
       a := E.X
      let E. := 3
       b := E.
    }
    assert(a == 34);
  }
}
```

The alternative to all this would be to actually change the meaning of dots in identifiers in inline assembly in general (always forcing them to refer to enums or scopes and disallowing identifiers ending in ``.``, etc.)

Reply:
Irrelevant to the behaviour we decide, I think this PR should have test cases for all. The problem we have is that dots don't have any meaning in the symbols and that creates a confusing situation.

Regards to the behaviour this is my personal opinion:
- Disallow identifiers starting with or ending in `.`
- The enum name would restrict that prefix to be used in other identifier names, e.g. if there is an enum `TxTypes` then any identifier starting with `TxTypes.` would be restricted to the enum only

I think this would help alleviate future problems.